hazelcast/src/main/java/com/hazelcast/query/impl/IndexImpl.java

```diff
@@ -79,7 +86,7 @@ public void clear() {
     }
 
     @Override
-    public void saveEntryIndex(QueryableEntry e) throws QueryException {
+    public void saveEntryIndex(QueryableEntry e, Object oldRecordValue) throws QueryException {
```

Comment:
yes. it could be. it should be probably renamed to `saveOrUpdate`. Again - I don't want to done too much changes here as it would cause conflicts with the multi-value branch.